### PR TITLE
Users can dispute transactions

### PR DESCRIPTION
**Pull request by garyhtou**
First, I added a partial for transactions without a linked object
image

On these `"no linked object" transactions`, `Stripe card transactions`, and `donations`, users are given an option to report the transaction. (this means you can not report ACHs, Checks, and Invoices — they're user-initiated so I figured there was no need)
image

When a user clicks "Report transaction form" link, the airtable form will open in a new tab (with some info prefilled)

### CODE REVIEW

**Comment by garyhtou** [comment]
This PR is a rework of https://github.com/hackclub/bank/pull/1788

**Comment by MelSmith104** [comment]
i love the feel of the airtable form right after they click "report this transaction" however the text itself seems a bit corporate - can we maybe have pass done on it to make it more line with the rest of the communication we have on Bank ?

File: Gemfile.lock [hunk 1]
```diff
@@ -280,7 +280,7 @@ GEM
       rake
     mini_magick (4.10.1)
     mini_mime (1.0.3)
-    mini_portile2 (2.5.1)
```
Comment (maxwofford) [inline Gemfile.lock]: Did you mean for this to be here?

File: Gemfile.lock [hunk 2]
```diff
@@ -301,8 +301,8 @@ GEM
     netrc (0.11.0)
     newrelic_rpm (6.15.0)
     nio4r (2.5.7)
-    nokogiri (1.11.5)
-      mini_portile2 (~> 2.5.0)
+    nokogiri (1.12.5)
+      mini_portile2 (~> 2.6.1)
```
Comment (maxwofford) [inline Gemfile.lock]: Should probably remove these– seems like it was accidentally committed

File: app/helpers/hcb_code_helper.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+require "cgi"
+
+module HcbCodeHelper
+  def report_transaction_airtable_form_url(embed: false, hcb_code: nil, user: nil)
+    # The airtable form is located within the Bank Promotions base
+    form_id = "shrf05pqZMlRs3gYJ"
```
Comment (maxwofford) [inline app/helpers/hcb_code_helper.rb]: If we're doing it this way, could i (a random person on the internet) just go to a transparency mode event and submit a fraud claim?

File: app/views/hcb_codes/_donation.html.erb [hunk 4]
```diff
@@ -3,7 +3,58 @@
   <%= link_to 'Issue refund', refund_donation_path(@hcb_code.donation), method: :post, class: 'btn bg-error' if @hcb_code.donation.deposited? %>
 <% end %>
 
-<article class="card pb0 mt3 mb3">
+<div class="mt2 mb3">
```
Comment (maxwofford) [inline app/views/hcb_codes/_donation.html.erb]: Was this intented for this PR?

File: app/views/hcb_codes/_rich_metadata.html.erb [hunk 5]
```diff
@@ -34,3 +34,7 @@
 <% end %>
 
 
+<!-- unknown (most likely a direct to bank account transaction) -->
```
Comment (maxwofford) [inline app/views/hcb_codes/_rich_metadata.html.erb]: Was this intentionally in this PR?

**Review by maxwofford — COMMENTED** [review]
This is so dope! Thanks for updating this. Couple comments:

Let's rename this feature "reported transactions" -> "fraud". Obv this is reductionist, but I think that's fine. If I go into the airtable and see a tab that says "fraud reports", I know what it means. If i look at the admin dashboard & see a task labeled "fraud" i know what it is.

Seems like there's some HCB code changes in here & diffs from https://github.com/hackclub/bank/pull/1812. Could you take out the extra stuff from this PR? If you need help with that let me know– i love fixing rebases.

Is this secure from random people on the internet & people on other bank events?

File: app/helpers/hcb_code_helper.rb (hunk 3 above)
Comment (garyhtou) [reply app/helpers/hcb_code_helper.rb]: It is linked under the HCB Code show page, which isn't visible in transparency. But theoretically, anyone can make a submission if they have the Airtable link

File: app/views/hcb_codes/_donation.html.erb (hunk 4 above)
Comment (garyhtou) [reply app/views/hcb_codes/_donation.html.erb]: This reorganizes the styling for the new "Report this Transaction" link

File: app/views/hcb_codes/_rich_metadata.html.erb (hunk 5 above)
Comment (garyhtou) [reply app/views/hcb_codes/_rich_metadata.html.erb]: This adds a card for transactions that cant be identified by Bank's transaction categorization system. There are likely to be transactions that go directly to our bank account (such as HQ's rippling payroll)

**Comment by garyhtou** [comment]
https://github.com/hackclub/airbridge/pull/49

**Comment by garyhtou** [comment]
## Partner Donation Screenshot
image

http://localhost:3000/hcb/HCB-201-59

Partner donations are donations that come from give butter

**Comment by garyhtou** [comment]
@maxwofford @MelSmith104 good to go!

**Comment by maxwofford** [comment]
@MelSmith104 let's review together today!

**Comment by MelSmith104** [comment]
awesome @garyhtou! i can't wait to review :) @maxwofford im free in a half hour!

**Comment by maxwofford** [comment]
Also see https://github.com/hackclub/airbridge/pull/50

**Pull request merged**